core/qbft/qbft.go

```diff
@@ -66,31 +72,38 @@ type Definition struct {
 // MsgType defines the QBFT message types.
 type MsgType int64
 
+// Note that message type ordering MUST not change, since it breaks backwards compatibility.
 const (
-	MsgPrePrepare MsgType = iota + 1
-	MsgPrepare
-	MsgCommit
-	MsgRoundChange
+	MsgUnknown     MsgType = 0
+	MsgPrePrepare  MsgType = 1
+	MsgPrepare     MsgType = 2
+	MsgCommit      MsgType = 3
+	MsgRoundChange MsgType = 4
+	msgSentinel    MsgType = 5
 )
 
+func (i MsgType) Valid() bool {
+	return i > MsgUnknown && i < msgSentinel
+}
+
 // Msg defines the inter process messages.
-type Msg struct {
+type Msg[I any, V Value[V]] interface {
```

Comment:
Changing message to an interface simplifies explicit justifications, since external buffer of signed messages isn't required.

Reply:
Also, here it is not clear to me what is I and what is V

Reply:
So, I guess V is the type of the value reaching a consensus on? But I?

Reply:
`I` is the instance of consensus as per the paper

Reply:
Ok, then